### PR TITLE
Add version HDF5 1.14.2

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class Hdf5(CMakePackage):
 
     # Odd versions are considered experimental releases
     # Even versions are maintenance versions
+    version("1.14.2", tag="hdf5-1_14_2", preferred=True)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Why are we using tags instead of tarballs?

Comment (lrknox) [reply var/spack/repos/builtin/packages/hdf5/package.py]: We wanted to check the CI testing before we were ready to publish the tarballs.  I'll update the PR with tarball and checksum in the next hour.

Comment (lrknox) [reply var/spack/repos/builtin/packages/hdf5/package.py]: downloading source tar.gz file now with sha256 checksum 1c342e634008284a8c2794c8e7608e2eaf26d01d445fb3dfd7f33cb2fb51ac53:
==> Installing hdf5-1.14.2-7isfblh67bj2xh7hr7wni43fknqzwrp2 [14/14]
==> No binary for hdf5-1.14.2-7isfblh67bj2xh7hr7wni43fknqzwrp2 found: installing from source
==> Fetching https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.2/src/hdf5-1.14.2.tar.gz

**Review by byrnHDF — APPROVED** [review]

**Comment by lrknox** [comment]
@alalazo I replaced tag with checksum for tarball, pulled latest develop, and all checks passed when they ran after some delay.  In the meantime develop changed again, creating conflicts, but it seems updating is not advised.  We hope to have this version in Spack for the E4S August release.  What do we need to do next?

**Review by kwryankrattiger — APPROVED** [review]
Looks good and it is tested in the Data Vis SDK stack

**Comment by kwryankrattiger** [comment]
@lrknox it doesn't have to be in this PR, but I think only putting the preferred flag on all of the versions defeats the purpose of the preferred flag. It is probably better to only tag it on the latest patch of each version.

**Comment by lrknox** [comment]
@kwryankrattiger, adding the preferred tag to all of the official releases was done to avoid defaulting to the experimental 1.13.x versions.  See comments in PR #29514.  Since the 1.13.x versions have been removed we can probably remove most of the preferred flags, but since we still have several supported versions it might be appropriate to keep the flag on at least the last 1.14.x, and 1.10.x versions.

**Pull request merged**